src/yaml/yamlUtils.ts

```diff
@@ -0,0 +1,257 @@
+                ':\r\n' +
+                document.getText().substr(lineOffsets[cursorPosition.line + 1] || document.getText().length)
+
+    // Non-empty line but missing colon, add colon to end of current line
+    } else {
+        modifiedDocText =
+            // tslint:disable-next-line: prefer-template
+            document.getText().substring(0, currentLineEnd) +
+                ':\r\n' +
+                document.getText().substr(lineOffsets[cursorPosition.line + 1] || document.getText().length)
+
+        // Starting pos is first non-space character
+        startPositionForInsertion.character = currentLine.indexOf(trimmedLine)
+    }
+
+    return {
+        modifiedDocText,
+        tempPositionForCompletions,
+        startPositionForInsertion,
+        endPositionForInsertion,
+        shouldPrependSpace
+    }
+}
+
+function processLineWithColon(document: TextDocument, cursorPosition: Position, currentLine: string, currentLineEnd: number): {
+    modifiedDocText: string,
+    tempPositionForCompletions: Position,
+    startPositionForInsertion: Position,
+    endPositionForInsertion: Position,
+    shouldPrependSpace: boolean
+} {
+    let modifiedDocText: string = document.getText()
+    let shouldPrependSpace: boolean = false
+
+    const tempPositionForCompletions: Position = {...cursorPosition}
+    const startPositionForInsertion: Position = {...cursorPosition}
+    const endPositionForInsertion: Position = {...cursorPosition}
+
+    const charNum = cursorPosition.character
+    const lineOffsets = getLineOffsets(document.getText())
+
+    // Current line has a colon, determine if cursor position is right or left of it
+    const colonIndex = currentLine.indexOf(':')
+
+    // Cursor is left of the colon.
+    if (charNum <= colonIndex) {
+        // Start position is first non-space character, ending position is left of colon.
+        const preColonTextTrimmed = currentLine.substring(0, colonIndex).trim()
+
+        // Only whitespace before the colon.
+        if (preColonTextTrimmed.length === 0) {
+            // Insert placeholder quotes and place cursor inside of them.
+            modifiedDocText =
+                // tslint:disable-next-line: prefer-template
+                document.getText().substring(0, lineOffsets[cursorPosition.line] + colonIndex) +
+                    "''" +
+                    document.getText().substr(lineOffsets[cursorPosition.line] + colonIndex)
+            startPositionForInsertion.character = colonIndex
+            endPositionForInsertion.character = colonIndex
+            tempPositionForCompletions.character = colonIndex + 1
+
+        // Only hyphen before the colon.
+        } else if (preColonTextTrimmed.length === 1 && preColonTextTrimmed.charAt(0) === '-') {
+            // Insert placeholder quotes and place cursor inside of them.
+            modifiedDocText =
+                // tslint:disable-next-line: prefer-template
```

Comment:
Same here, let's actually make it clearer and use template string with properly named variable constants above.

Reply:
Updated to be template string.